scripts/llnl/common_build_functions.py

```diff
@@ -637,9 +637,26 @@ def get_build_dir(prefix, host_config):
     return pjoin(prefix, "build-" + host_config_root)
 
 
+_repo_dir = ""
 def get_repo_dir():
-    script_dir = os.path.dirname(os.path.realpath(__file__))
-    return os.path.abspath(pjoin(script_dir, "../.."))
+    """ Return absolute path to Smith or Serac repo """
+    # Only set var once
+    global _repo_dir
+    if not _repo_dir:
+        # Set repo dir to be relative to this script
+        _repo_dir = os.path.abspath(pjoin(get_script_dir(), "../.."))
+
+        # Check the parent dir for an uberenv config with a package_name "smith".
+        # This means Serac is a submodule. Change the repo dir to be parent dir (i.e.
+        # Smith repo dir)
+        smith_uberenv_config_path = pjoin(_repo_dir, "../.uberenv_config.json")
+        if os.path.exists(smith_uberenv_config_path):
+            with open(smith_uberenv_config_path) as json_file:
```

Comment:
Should use get_project_name() here...

Reply:
The problem with that is `get_project_name()` depends on `get_repo_dir()`. I could move the `.uberenv_config.json` query to a separate function though.